packages/buidler-etherscan/src/etherscan/EtherscanService.ts

```diff
@@ -1,74 +1,120 @@
 import { NomicLabsBuidlerPluginError } from "@nomiclabs/buidler/plugins";
-import request from "request-promise";
 
-import { EtherscanRequestParameters } from "./EtherscanVerifyContractRequest";
+import { pluginName } from "../pluginContext";
 
-async function delay(ms: number): Promise<void> {
+import {
+  EtherscanCheckStatusRequest,
+  EtherscanVerifyRequest,
+} from "./EtherscanVerifyContractRequest";
+
+export async function delay(ms: number): Promise<void> {
   return new Promise((resolve) => setTimeout(resolve, ms));
 }
 
+// Used for polling the result of the contract verification.
+const verificationIntervalMs = 3000;
+
 export async function verifyContract(
-  url: string,
-  req: EtherscanRequestParameters
+  url: URL,
+  req: EtherscanVerifyRequest
 ): Promise<EtherscanResponse> {
+  const { default: fetch } = await import("node-fetch");
+  // The API expects the whole request in the search parameters.
```

Comment:
Are you sure about this? The previous version was sending them in the body of a POST request. Maybe the API changed since then, but if that was the case I think we'd know.